src/main/java/com/ericsson/ei/handlers/ObjectHandler.java

```diff
@@ -155,22 +161,26 @@ public String findObjectById(String id) {
         return objects;
     }
 
-    public JsonNode prepareDocumentForInsertion(String id, String object) {
-        ObjectMapper mapper = new ObjectMapper();
-        try {
-            String docStr = "{\"_id\": \"" + id + "\"}";
-            JsonNode jsonNodeNew = mapper.readValue(docStr, JsonNode.class);
-
-            JsonNode jsonNode = mapper.readValue(jsonNodeNew.toString(), JsonNode.class);
-            ObjectNode objNode = (ObjectNode) jsonNode;
-            objNode.set("aggregatedObject", mapper.readTree(object));
-
-            return jsonNode;
-        } catch (Exception e) {
-            log.error(e.getMessage(), e);
-        }
-        return null;
-    }
+	public BasicDBObject prepareDocumentForInsertion(String id, String object) {
+		try {
+			Date date = new Date();
```

Comment:
put date creation as a static method in another class and call this method in the if statement below since it is only needed if getTtl is greater than 0.

Reuse this time function in the other places where we add TTL time.